clusterloader2/pkg/measurement/common/scheduler_latency.go

```diff
@@ -134,37 +146,90 @@ func (*schedulerLatencyMeasurement) String() string {
 	return schedulerLatencyMetricName
 }
 
-func (s *schedulerLatencyMeasurement) resetSchedulerMetrics(c clientset.Interface, host, provider, masterName string, masterRegistered bool) error {
-	_, err := s.sendRequestToScheduler(c, "DELETE", host, provider, masterName, masterRegistered)
-	if err != nil {
-		return err
+// Helper function to substract two histograms
+func histogramSub(finalHist, initialHist *measurementutil.Histogram) *measurementutil.Histogram {
+	for k := range finalHist.Buckets {
+		finalHist.Buckets[k] = finalHist.Buckets[k] - initialHist.Buckets[k]
 	}
-	return nil
+	return finalHist
 }
 
 // Retrieves scheduler latency metrics.
 func (s *schedulerLatencyMeasurement) getSchedulingLatency(c clientset.Interface, host, provider, masterName string, masterRegistered bool) ([]measurement.Summary, error) {
 	result := schedulingMetrics{
 		FrameworkExtensionPointDuration: make(map[string]*measurementutil.LatencyMetric),
 	}
-	data, err := s.sendRequestToScheduler(c, "GET", host, provider, masterName, masterRegistered)
+	for _, ePoint := range extentionsPoints {
+		result.FrameworkExtensionPointDuration[ePoint] = &measurementutil.LatencyMetric{}
+	}
+
+	schedulerMetrics, err := s.getSchedulingMetrics(c, host, provider, masterName, masterRegistered)
 	if err != nil {
 		return nil, err
 	}
 
-	samples, err := measurementutil.ExtractMetricSamples(data)
+	if s.initialLatency == nil {
+		return nil, fmt.Errorf("initial values are not available")
+	}
+	schedulerMetrics.preemptionEvaluationHist = histogramSub(schedulerMetrics.preemptionEvaluationHist, s.initialLatency.preemptionEvaluationHist)
```

Comment:
Let's make one more step and add a method on schedulerLatencyMetriic like:

```
func (m *schedulerLatencyMetric) substract(sub *schedulerLatencyMetric) {
  m.preemptionEvaluationHist = histogramSub(m.preemptionEvaluationHist, sub.preemptionEvaluationHist)
  ...
}
```

and use it here

Reply:
add the new method